core/src/fwd/Kokkos_Fwd_CUDA.hpp

```diff
@@ -61,6 +61,18 @@ void cuda_prefetch_pointer(const ExecSpace& /*space*/, const void* /*ptr*/,
 void cuda_prefetch_pointer(const Cuda& space, const void* ptr, size_t bytes,
                            bool to_device);
 
+template <typename ConfigT>
+struct SpacePriority<Kokkos::Cuda, ConfigT>
+    : std::integral_constant<int, 50> {};
+
+template <>
+struct SpacePriority<Kokkos::Cuda, Kokkos::HostParallelPriorityConfig>
+    : std::integral_constant<int, 999> {};
+
+template <>
+struct SpaceProperty<CUDA_idx> {
```

Comment:
Where does this come from?

Reply:
I'll post the original prototype, which might help demonstrate why it's done this way (and allow you to play around with it if you want)